src/main/kotlin/brs/db/sql/SqlSubscriptionStore.kt

```diff
@@ -19,28 +18,69 @@ internal class SqlSubscriptionStore(private val dp: DependencyProvider) : Subscr
         }
     }
 
-    override val subscriptionTable: VersionedEntityTable<Subscription>
+    override val subscriptionTable: MutableEntityTable<Subscription>
 
     init {
         subscriptionTable =
-            object : SqlVersionedEntityTable<Subscription>(
+            object : SqlMutableEntityTable<Subscription>(
                 SUBSCRIPTION,
                 SUBSCRIPTION.HEIGHT,
                 SUBSCRIPTION.LATEST,
                 subscriptionDbKeyFactory,
                 dp
             ) {
                 override val defaultSort = listOf(
-                    table.field("time_next", Int::class.java).asc(),
-                    table.field("id", Long::class.java).asc()
+                    SUBSCRIPTION.TIME_NEXT.asc(),
+                    SUBSCRIPTION.ID.asc()
                 )
 
-                override fun load(ctx: DSLContext, record: Record): Subscription {
-                    return sqlToSubscription(record)
-                }
+                override fun load(record: Record) = Subscription(
+                    record.get(SUBSCRIPTION.SENDER_ID),
+                    record.get(SUBSCRIPTION.RECIPIENT_ID),
+                    record.get(SUBSCRIPTION.ID),
+                    record.get(SUBSCRIPTION.AMOUNT),
+                    record.get(SUBSCRIPTION.FREQUENCY),
+                    record.get(SUBSCRIPTION.TIME_NEXT),
+                subscriptionDbKeyFactory.newKey(record.get(SUBSCRIPTION.ID)))
+
+                private val upsertColumns = listOf(
+                    SUBSCRIPTION.ID,
+                    SUBSCRIPTION.SENDER_ID,
+                    SUBSCRIPTION.RECIPIENT_ID,
+                    SUBSCRIPTION.AMOUNT,
+                    SUBSCRIPTION.FREQUENCY,
+                    SUBSCRIPTION.TIME_NEXT,
+                    SUBSCRIPTION.HEIGHT,
+                    SUBSCRIPTION.LATEST
+                )
+                private val upsertKeys = upsertColumns // TODO do we really need every column unique? what's the point?
```

Comment:
Could be more than one record SENDER_ID-RECIPIENT_ID on one HEIGHT? - I think SENDER_ID-RECIPIENT_ID-HEIGHT - is unique.

Reply:
subscriptions should be never updated, only inserts, iiiif get rid upsert .... xD

Reply:
if see on code above, there unique on transaction_id-height...need to think. Maybe enough only transaction_id (which id)? Transaction_id is always unique per table? Can we ident row by transaction_id? I think yes but not sure.